FreeRTOS-Plus/Demo/coreSNTP_Windows_Simulator/DemoTasks/SNTPClientTask.c

```diff
@@ -1379,6 +1431,45 @@ static void closeUdpSocket( Socket_t * pSocket )
 
 /*-----------------------------------------------------------*/
 
+static bool calculateBackoffForNextPoll( bool firstBackoffAttempt,
+                                         uint32_t minPollPeriod,
+                                         uint32_t * pPollPeriod )
+{
+    uint16_t newPollPeriod = 0U;
+    BackoffAlgorithmStatus_t status;
+    static BackoffAlgorithmContext_t backoffParams;
```

Comment:
I feel like this does not need to be static. It is no longer safe to use if you need to calculate backoff while something else also needs to calculate backoff.

Reply:
I was trying to encapsulate all the types of BackoffAlgorithm algorithm within this function. I agree that it makes the re-use of this function not feasible. So, I have moved the declaration of the `BackoffAlgorithmContext_t` context out and made it a parameter to the function